tests: kernel: mslab: Add timeout parameter

golowanow opened the pull request:
Add TEST_ALLOC_TIMEOUT_MS config parameter instead of hardcoded timeout at kernel.memory_slabs test to allow its customization e.g. on slow simulated platforms.

golowanow committed:
tests: kernel: mslab: Add timeout parameter

Add TEST_ALLOC_TIMEOUT_MS config parameter instead of hardcoded
timeout at kernel.memory_slabs test to allow its customization
on slow simulated platforms.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

andyross commented on tests/kernel/mem_slab/mslab/src/main.c:
```diff
@@ -247,9 +252,10 @@ ZTEST(memory_slab_1cpu, test_mslab)
 	zassert_equal(0, ret_value,
-		      "Failed k_mem_slab_alloc, ret_value %d\n", ret_value);
+		      "Failed k_mem_slab_alloc in %d msec., ret_value %d\n",
+		      K_MSEC(50 * (TIMEOUT_MULTIPLIER)), ret_value);
```
I... don't understand this?  The test doesn't care about "50ms", it's not looking for a timeout.  It wants the call to **succeed**, and the fact that it times out is an error.  So why make this platform dependent?  Just set it to 100ms for everyone (or K_FOREVER logically, though I'm guessing that the reason this is a timeout is some kind of coverage requirement).

golowanow replied:
yes, this test expects to get a memory block in time which depends on another thread's k_sem_take() and k_mem_slab_free() timings, so to me it looks unsafe to just increase the timeout here because of some particular platform potentially affecting other platforms.
imo, the proposed change with config parameter keeps this test backward compatible and prevents oversight if k_sem_* or k_mem_slab_ delays are really going bad somewhere.

andyross replied:
How would it be unsafe?  Existing platforms are **passing** this test, so we know they can get the result back in 50ms.   So they can definitely get it back in less than 100ms.

Again, it's just not a test of allocator speed.  Let's not add platform dependence needlessly.  Can we change this to 100ms globally?

golowanow replied:
> How would it be unsafe? Existing platforms are **passing** this test, so we know they can get the result back in 50ms. So they can definitely get it back in less than 100ms.
> 
well, as a 'false positive', e.g. if some existing platform get this test passing with 100ms timeout whereas its usual 40 ms unexpectedly raised to 80ms.

> Again, it's just not a test of allocator speed. Let's not add platform dependence needlessly.
> Can we change this to 100ms globally?

in our case this timeout should be 150ms to pass.

andyross reviewed (changes requested):
Same general comments from #58862 , but this one is weird enough that I think it rises to a -1 from me pending explanation.

golowanow committed:
tests: kernel: mslab: intel_ish_5_8_0: extend timeout

Extend test's timeout for k_mem_slab_alloc() call on intel_ish_5_8_0
simulated platform.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
@andyross could you please review this again ? 
After a while it is reopened -- the timeout parameter with larger value is needed to run `kernel.memory_slabs` on `intel_ish_5_8_0` simulated board.

nashif reviewed (changes requested):
fully agree with @andyross. We should not have platform specifics like this in the tests, I would just increase value for everyone.

golowanow commented:
closed in favor of #88696
